aws/resource_aws_api_gateway_gateway_response.go

```diff
@@ -0,0 +1,143 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayGatewayResponse() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayGatewayResponsePut,
+		Read:   resourceAwsApiGatewayGatewayResponseRead,
+		Update: resourceAwsApiGatewayGatewayResponsePut,
+		Delete: resourceAwsApiGatewayGatewayResponseDelete,
+
+		Schema: map[string]*schema.Schema{
+			"rest_api_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"response_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"status_code": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"response_templates": {
+				Type:     schema.TypeMap,
+				Elem:     schema.TypeString,
+				Optional: true,
+			},
+
+			"response_parameters": {
+				Type:     schema.TypeMap,
+				Elem:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayGatewayResponsePut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	templates := make(map[string]string)
+	for k, v := range d.Get("response_templates").(map[string]interface{}) {
```

Comment:
Shouldn't this be also wrapped in `GetOk()` like the other optional field, as it's optional? Otherwise I think we might crash here when casting `nil` to a map.